src/FTPClientWrapperSSH.cpp

```diff
@@ -246,8 +246,9 @@ int FTPClientWrapperSSH::ReceiveFile(HANDLE hFile, const char * ftpfile) {
 	ssize_t retcode = 0;
 	int res = TRUE;
 	sftp_file sfile = NULL;
-	const int bufsize = 4096;
-	char buf[bufsize];
+	const int bufsize = 128 * 1024;

```

Comment:
Instead of const int, probably const size_t is the expected type for the element count (https://en.cppreference.com/w/cpp/container/vector/vector).

Have you tested also with 16kB from stack? For this std:array could be used then.

Sorry for being not as fast as you modified it.

Reply:
Plain C array sizes and and `sftp_read()` also expects `size_t` so it should technically always have been `size_t`, but for these values it makes no difference. It's not being built on any platform with 16 bit integers. I just wanted to make the change as minimal as possible.

As for 16 KiB, I did test that. Any value lower than around 100 KiB was noticeably slower. I'm still a bit puzzled why much bigger buffers did not speed things more up, but maybe that's a libssh v1 vs. v2 issue.